### PR TITLE
Code Cleanup.

### PR DESCRIPTION
**Pull request by shahabkondri**

### CODE REVIEW

File: src/main/java/org/apache/commons/compress/archivers/ar/ArArchiveEntry.java [hunk 1]
```diff
@@ -179,12 +179,7 @@ public boolean equals(final Object obj) {
...
         if (name == null) {
-            if (other.name != null) {
-                return false;
-            }
-        } else if (!name.equals(other.name)) {
-            return false;
-        }
-        return true;
+            return other.name == null;
+        } else return name.equals(other.name);
```
Comment (bodewig) [inline src/main/java/org/apache/commons/compress/archivers/ar/ArArchiveEntry.java]: please add braces for the else block.

File: src/main/java/org/apache/commons/compress/archivers/cpio/CpioArchiveEntry.java [hunk 2]
```diff
@@ -886,12 +886,7 @@ public boolean equals(final Object obj) {
...
         if (name == null) {
-            if (other.name != null) {
-                return false;
-            }
-        } else if (!name.equals(other.name)) {
-            return false;
-        }
-        return true;
+            return other.name == null;
+        } else return name.equals(other.name);
```
Comment (bodewig) [inline src/main/java/org/apache/commons/compress/archivers/cpio/CpioArchiveEntry.java]: please add braces for the else block.

File: src/main/java/org/apache/commons/compress/archivers/dump/DumpArchiveEntry.java [hunk 3]
```diff
@@ -418,12 +418,8 @@ public boolean equals(final Object o) {
...
         // summary is always null right now, but this may change some day
-        if ((summary == null && rhs.summary != null) // NOSONAR
-            || (summary != null && !summary.equals(rhs.summary))) { // NOSONAR
-            return false;
-        }
-
-        return true;
+        return (summary != null || rhs.summary == null) // NOSONAR
+                && (summary == null || summary.equals(rhs.summary));
     }
```
Comment (bodewig) [inline src/main/java/org/apache/commons/compress/archivers/dump/DumpArchiveEntry.java]: I find this version a lot harder to read than the original expression. I'm totally fine with getting rid of the `if`, but please keep the original logic.

**Comment by shahabkondri** [comment]
Thanks for reviewing. 
The suffix L is preferred, because the letter l (ell) is often hard to distinguish from the digit 1 (one).

File: src/main/java/org/apache/commons/compress/archivers/dump/DumpArchiveEntry.java (hunk 3 above)
Comment (shahabkondri) [reply src/main/java/org/apache/commons/compress/archivers/dump/DumpArchiveEntry.java]: I've double checked and reverted to the original logic.

**Comment by bodewig** [comment]
merged, many thanks!

**Pull request closed without merging**